test_project/test_app/tests/factories.py

```diff
@@ -22,11 +23,20 @@ def _create(cls, model_class, *args, **kwargs):
         user = super(UserFactory, cls)._create(model_class, *args, **kwargs)
         # Force save for post_save signal to create auth client
         user.save()
-        AccessToken.objects.create(user=user,
-                                   application=user.oauth2_provider_application.first(),
-                                   token='token{}'.format(user.id),
-                                   expires=now() + datetime.timedelta(days=1)
-                                   )
+        oauth_toolkit_version = [int(num) for num in oauth2_provider.__version__.split('.')]
+        # If we're using version 0.8.0 or higher
+        if oauth_toolkit_version[0] >= 0 and oauth_toolkit_version[1] >= 8:
+            AccessToken.objects.create(user=user,
+                                       application=user.oauth2_provider_application.first(),
```

Comment:
Is the only difference here the `application`? Might be good to just set the variable in the if/else logic then only have one line that creates the `AccessToken`